src/routes/ChatRoom/ChatRoom.connector.js

```diff
@@ -0,0 +1,224 @@
+import { push } from 'connected-react-router'
+import { get, includes, isEmpty } from 'lodash/fp'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
+import { createSelector as ormCreateSelector } from 'redux-orm'
+import {
+  addAttachment,
+  clearAttachments,
+  getAttachments
+} from 'components/AttachmentManager/AttachmentManager.store'
+import {
+  MODULE_NAME,
+  FETCH_LINK_PREVIEW,
+  pollingFetchLinkPreview,
+  removeLinkPreview,
+  clearLinkPreview,
+  getLinkPreview
+} from 'components/PostEditor/PostEditor.store'
+import { updateUserSettings } from 'routes/UserSettings/UserSettings.store'
+import changeQuerystringParam from 'store/actions/changeQuerystringParam'
+import createPost from 'store/actions/createPost'
+import fetchGroupTopic from 'store/actions/fetchGroupTopic'
+import fetchPosts from 'store/actions/fetchPosts'
+import fetchTopic from 'store/actions/fetchTopic'
+import respondToEvent from 'store/actions/respondToEvent'
+import toggleGroupTopicSubscribe from 'store/actions/toggleGroupTopicSubscribe'
+import updateGroupTopicLastReadPost from 'store/actions/updateGroupTopicLastReadPost'
+import updatePost from 'store/actions/updatePost'
+import { FETCH_POSTS, FETCH_TOPIC, FETCH_GROUP_TOPIC } from 'store/constants'
+import orm from 'store/models'
+import presentPost from 'store/presenters/presentPost'
+import getGroupForCurrentRoute from 'store/selectors/getGroupForCurrentRoute'
+import getGroupTopicForCurrentRoute from 'store/selectors/getGroupTopicForCurrentRoute'
+import getMe from 'store/selectors/getMe'
+import getMyMemberships from 'store/selectors/getMyMemberships'
+import getQuerystringParam from 'store/selectors/getQuerystringParam'
+import { getHasMorePosts, getPostResults, getTotalPosts } from 'store/selectors/getPosts'
+import getRouteParam from 'store/selectors/getRouteParam'
+import getTopicForCurrentRoute from 'store/selectors/getTopicForCurrentRoute'
+import isPendingFor from 'store/selectors/isPendingFor'
+import { postUrl } from 'util/navigation'
+
+// selectors
+export const getPosts = ormCreateSelector(
+  orm,
+  getPostResults,
+  (session, results) => {
+    if (isEmpty(results)) return null
+    if (isEmpty(results.ids)) return []
+    return session.Post.all()
+      .filter(x => includes(x.id, results.ids))
+      .orderBy(p => Number(p.id))
+      .toModelArray()
+      .map(p => presentPost(p))
+  }
+)
+
+const NUM_POSTS_TO_LOAD = 25
+
+export function mapStateToProps (state, props) {
```

Comment:
So you have a connector for Chatroom, using the class component style of redux, feeding into Chatroom.js, which is a functional component? Mixing these patterns has only ever given me huge grief in the past. I really recommend avoiding the use of the connector pattern directly feeding into a functional component.

Reply:
In agreement on this. I think we should consider `.connector.js` files deprecated in the case of new components. I haven't personally had any issue with having a connector HOC wrapping a functional component which also uses redux hooks, but I do think it's hard to track/understand what's going on.

Reply:
yeah I think you are right, though actually I think ChatRoom should not have been a functional component, its too complex and the hooks are super confusing. not sure what to do at this point